### PR TITLE
sunshine: use cuda compatible libstdc++ when building with cuda support

### PR DESCRIPTION
**Pull request by icewind1991**
###### Description of changes

Fix building sunshine with cuda support, solution inspired by https://github.com/NixOS/nixpkgs/pull/223664/files#diff-49ad394a5939e5901faf2863d126d07c1f6e8fd96cba2e1773aef49203b8cd02R39

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/sunshine/default.nix [hunk 1]
```diff
@@ -41,8 +41,9 @@ let
     url = "https://raw.githubusercontent.com/LizardByte/Sunshine/6702802829869547708dfec98db5b8cbef39be89/cmake/FindFFMPEG.cmake";
     sha256 = "sha256:1hl3sffv1z8ghdql5y9flk41v74asvh23y6jmaypll84f1s6k1xa";
   };
+  stdenv' = if cudaSupport then cudaPackages.backendStdenv else stdenv;
```
Comment (DorZion) [inline pkgs/servers/sunshine/default.nix:44]: Maybe rename to `fixedStdenv`?

**Review by DorZion — APPROVED** [review]
LGTM, did the same fix locally when used `nixos-unstable` channel.

**Comment by icewind1991** [comment]
I've rebased the PR, but I no longer have a NixOS system with an nvidia card so I can't perform real testing

**Comment by Nemo157** [comment]
I've tested doing the same change via a local `.override { cudaSupport = true; stdenv = pkgs-final.cudaPackages.backendStdenv; }` and it works for me, fixing nvenc support.

**Comment by devusb** [comment]
completed as part of https://github.com/NixOS/nixpkgs/pull/293938

**Pull request closed without merging**